core/trino-main/src/main/java/io/trino/sql/planner/optimizations/ReplicateJoinAndSemiJoinInDelete.java

```diff
@@ -41,6 +46,20 @@ public PlanNode optimize(PlanNode plan, Session session, TypeProvider types, Sym
     {
         private boolean isDeleteQuery;
 
+        @Override
+        public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
+        {
+            PlanNode leftSourceRewritten = context.rewrite(node.getLeft(), context.get());
+
+            JoinNode rewrittenNode = (JoinNode) node.replaceChildren(ImmutableList.of(leftSourceRewritten, node.getRight()));
+            Type joinType = rewrittenNode.getType();
+            if (isDeleteQuery && (joinType == INNER || joinType == LEFT)) {
```

Comment:
i guess this should be:

```
if (isDeleteQuery) {
  verify(joinType == INNER || joinType == LEFT
```

Reply:
what about UPDATE?

Reply:
UPDATE can be applied in a follow up PR.

Reply:
> i guess this should be:
> 
> ```
> if (isDeleteQuery) {
>   verify(joinType == INNER || joinType == LEFT
> ```

has this been resolved?
or, am i being wrong here?